src/IssueLabeler/AzureSdkIssueLabelerService.cs

```diff
@@ -51,23 +106,71 @@ public async Task<ActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "P
-                    issue.RepositoryName,
+                    predictionRepositoryName,
                     issue.RepositoryOwnerName);
 
                 if (predictions.Count < 2)
                 {
+                    log.LogInformation($"No labels were predicted for {issue.RepositoryName} using the `{predictionRepositoryName}` model for issue #{issue.IssueNumber}.");
                     return EmptyResult;
                 }
 
+                log.LogInformation($"Labels were predicted for {issue.RepositoryName} using the `{predictionRepositoryName}` model for issue #{issue.IssueNumber}.  Using: [{predictions[0]}, {predictions[1]}].");
                 return new JsonResult(new PredictionResponse(predictions));
             }
             catch (Exception ex)
             {
-                log.LogError($"Error querying predictions: {ex.Message}{Environment.NewLine}\t{ex}{Environment.NewLine}");
+                log.LogError($"Error querying predictions for {issue.RepositoryName} using the `{predictionRepositoryName}` model for issue #{issue.IssueNumber}: {ex.Message}{Environment.NewLine}\t{ex}{Environment.NewLine}");
                 return EmptyResult;
             }
         }
 
+        private async Task Initialize(IConfiguration config, IModelHolderFactoryLite modelHolderFactory)
+        {
+            // Initialize the models to use for prediction.
+
+            var owner = config["RepoOwner"];
+            var repos = config["RepoNames"];
+
+            if (repos != null)
+            {
+                foreach (var repo in repos.Split(';', StringSplitOptions.RemoveEmptyEntries))
+                {
+                    var blobConfig = $"IssueModel:{repo}:BlobConfigNames";
+
+                    if (!string.IsNullOrEmpty(config[blobConfig]))
+                    {
+                        var blobConfigs = config[blobConfig].Split(';', StringSplitOptions.RemoveEmptyEntries);
+                        foreach (var blobConfigName in blobConfigs)
+                        {
+                            await modelHolderFactory.CreateModelHolder(owner, repo, blobConfigName).ConfigureAwait(false);
+                        }
+                    }
+                    else
+                    {
+                        await modelHolderFactory.CreateModelHolder(owner, repo).ConfigureAwait(false);
+                    }
+                }
+            }
+
+            // Initialize the set of repositories that use the common model.
+
+            var commonModelRepos = Config["ReposUsingCommonModel"];
+
+            if (!string.IsNullOrEmpty(commonModelRepos))
+            {
+                foreach (var repo in commonModelRepos.Split(';', StringSplitOptions.RemoveEmptyEntries))
+                {
+                    CommonModelRepositories.Add(repo);
+                }
+            }
+        }
+
+        private string TranslateRepoName(string repoName) =>
+            CommonModelRepositories.Contains(repoName)
```

Comment:
is contains ever going to be a problem where a repo is a substring of another? Probably rare, but `-pr` repos are one example I can think of.

Reply:
If we get that far, we'll probably want a full-blown map, as the "pr" variants would use the model from the public version of those repositories.